pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -486,10 +486,13 @@ func (vs *vSphereVMProvider) vmCreateDoPlacement(
 	vcClient *vcclient.Client,
 	createArgs *VMCreateArgs) error {
 
-	placementConfigSpec := virtualmachine.CreateConfigSpecForPlacement(
+	placementConfigSpec, err := virtualmachine.CreateConfigSpecForPlacement(
 		vmCtx,
 		createArgs.ConfigSpec,
 		createArgs.StorageClassesToIDs)
+	if err != nil {
+		return err
+	}
```

Comment:
Consider adding `conditions.MarkFalse(vmCtx.VM, vmopv1.VirtualMachineConditionPlacementReady, "NotReady", err.Error())` here?

Reply:
You are quite kind to not just state that I missed this :)